### PR TITLE
Tray context menu improvements

### PR DESCRIPTION
**Pull request by raksooo**
This PR improves the context menu by:
* Move code that creates the menu to `tray-icon-controller.ts`
* Change layout of context menu to not change text and action of items and instead just change `enabled`
* Update context menu when account, daemon connection or language changes.

Git checklist:

* [x] Describe the change in **`CHANGELOG.md`** under the `[Unreleased]` header.
* [x] Check that commits follow the [Mullvad coding guidelines](https://github.com/mullvad/coding-guidelines)

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)

### CODE REVIEW

**Review by dlon — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: all files reviewed, 3 unresolved discussions (waiting on @raksooo)

___
*[CHANGELOG.md, line 77 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2FPM2Qq82Tcc3P4r:-Mtc2FPN6d8Re8Opr_6H:b-cbczcp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/CHANGELOG.md#L77)):*
> ```Markdown
> - Prevent gRPC from trying to run the app-daemon IPC communication through a HTTP proxy when the
>   environment variable `http_proxy` is set. This caused the app to fail to connect to the daemon.
> - Improved tray context menu behavior. There are now separate items for connect, disconnect and
> ```

Nit: Does this belong under `Changed`?
___
*[gui/src/shared/connect-helper.ts, line 10 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2sO35lZ1PYlaXIIt:-Mtc2sO45m_a7uFUi260:b-qqkrda) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/gui/src/shared/connect-helper.ts#L10)):*
> ```XML
>   return (
>     connectedToDaemon &&
>     accountToken !== undefined &&
> ```

Could it be simplified to this?

```
  return (
    connectedToDaemon &&
    accountToken &&
    (tunnelState === 'disconnected' || tunnelState === 'disconnecting' || tunnelState === 'error')
```
___
*[gui/src/shared/connect-helper.ts, line 24 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc3cjgED66MzWYz3Bk:-Mtc3cjgED66MzWYz3Bl:b-7uab7x) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/gui/src/shared/connect-helper.ts#L24)):*
> ```XML
>     connectedToDaemon &&
>     accountToken !== undefined &&
>     accountToken !== '' &&
> ```

See the above comment.

**Review by raksooo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: all files reviewed, 3 unresolved discussions (waiting on @dlon and @raksooo)

___
*[CHANGELOG.md, line 77 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2FPM2Qq82Tcc3P4r:-MtlQu5x5y5LpXXvbBUC:b6zxage) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/CHANGELOG.md#L77)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Nit: Does this belong under `Changed`?
</blockquote></details>

Are you saying it should be under `Changed` instead? The old behavior had flaws and bugs so IMO it belongs under `Fixed`. Maybe I should reword it to sound more like a fix?
___
*[gui/src/shared/connect-helper.ts, line 10 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2sO35lZ1PYlaXIIt:-MtlRB7D0TULr4MDVhjJ:b-6wllec) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/gui/src/shared/connect-helper.ts#L10)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Could it be simplified to this?

```
  return (
    connectedToDaemon &&
    accountToken &&
    (tunnelState === 'disconnected' || tunnelState === 'disconnecting' || tunnelState === 'error')
```
</blockquote></details>

The problem is that it would result in the type `boolean | string` for this return value since:
```
string && boolean => string | boolean
"" && true => ""
```
And a boolean is expected as result. To work around this we would have to do something like `!!accountToken` or `new Boolean(accountToken)` but I'm not a big fan of any of those.

Wouldn't it be a better idea to send `None` for account number when not logged in? Then it would just be the `accountToken !== undefined` here. But I'm not suggesting we change this now :P

**Review by dlon — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: all files reviewed, 1 unresolved discussion (waiting on @raksooo)

___
*[CHANGELOG.md, line 77 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2FPM2Qq82Tcc3P4r:-MtlYnl633V-V2gc13mz:byiyd79) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/CHANGELOG.md#L77)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

Are you saying it should be under `Changed` instead? The old behavior had flaws and bugs so IMO it belongs under `Fixed`. Maybe I should reword it to sound more like a fix?
</blockquote></details>

Alright. This is fine, but it could say something about what was actually not working correctly before.
___
*[gui/src/shared/connect-helper.ts, line 10 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2sO35lZ1PYlaXIIt:-MtlYL8N6qOibO4BxLgk:b-f1h1oo) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/gui/src/shared/connect-helper.ts#L10)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

The problem is that it would result in the type `boolean | string` for this return value since:
```
string && boolean => string | boolean
"" && true => ""
```
And a boolean is expected as result. To work around this we would have to do something like `!!accountToken` or `new Boolean(accountToken)` but I'm not a big fan of any of those.

Wouldn't it be a better idea to send `None` for account number when not logged in? Then it would just be the `accountToken !== undefined` here. But I'm not suggesting we change this now :P
</blockquote></details>

Ah, okay. Keep it like this then.

**Review by dlon — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-:-MtlZSmH8FRL09xi1F__:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @raksooo)

**Review by dlon — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: all files reviewed, 1 unresolved discussion (waiting on @raksooo)

___
*[gui/src/shared/connect-helper.ts, line 10 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2sO35lZ1PYlaXIIt:-Mtm2Okd73OPJohKWU1B:b-qh5vc8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/gui/src/shared/connect-helper.ts#L10)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Ah, okay. Keep it like this then.
</blockquote></details>

As for sending `None`, IIRC gRPC makes no distinction between empty strings and undefined strings. It could be wrapped in another message, I suppose. But I think this is fine.

**Review by raksooo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: 3 of 4 files reviewed, all discussions resolved (waiting on @dlon)

___
*[CHANGELOG.md, line 77 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266#-Mtc2FPM2Qq82Tcc3P4r:-MtmAl6x1bC4PGAvrKDW:brkno94) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/fa3bcdb5107510f69315e255aad57af5507af5ad/CHANGELOG.md#L77)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Alright. This is fine, but it could say something about what was actually not working correctly before.
</blockquote></details>

I agree, done.

**Review by dlon — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3266)* status: :shipit: complete! all files reviewed, all discussions resolved
